app/src/main/kotlin/com/google/samples/apps/nowinandroid/ui/interests2pane/InterestsListDetailScreen.kt

```diff
@@ -93,70 +99,139 @@ internal fun InterestsListDetailScreen(
             },
         ),
     )
-    BackHandler(listDetailNavigator.canNavigateBack()) {
-        listDetailNavigator.navigateBack()
+    val coroutineScope = rememberCoroutineScope()
+
+    val paneExpansionState = rememberPaneExpansionState(
+        anchors = listOf(
+            PaneExpansionAnchor.Proportion(0f),
+            PaneExpansionAnchor.Proportion(0.5f),
+            PaneExpansionAnchor.Proportion(1f),
+        ),
+    )
+
+    ThreePaneScaffoldPredictiveBackHandler(
+        listDetailNavigator,
+        BackNavigationBehavior.PopUntilScaffoldValueChange,
+    )
+    BackHandler(
+        paneExpansionState.currentAnchor == PaneExpansionAnchor.Proportion(0f) &&
+            listDetailNavigator.isListPaneVisible() &&
+            listDetailNavigator.isDetailPaneVisible(),
+    ) {
+        coroutineScope.launch {
+            paneExpansionState.animateTo(PaneExpansionAnchor.Proportion(1f))
+        }
     }
 
     var nestedNavHostStartRoute by remember {
         val route = selectedTopicId?.let { TopicRoute(id = it) } ?: TopicPlaceholderRoute
         mutableStateOf(route)
     }
-    var nestedNavKey by rememberSaveable(
-        stateSaver = Saver({ it.toString() }, UUID::fromString),
-    ) {
-        mutableStateOf(UUID.randomUUID())
-    }
-    val nestedNavController = key(nestedNavKey) {
-        rememberNavController()
-    }
 
     fun onTopicClickShowDetailPane(topicId: String) {
         onTopicClick(topicId)
-        if (listDetailNavigator.isDetailPaneVisible()) {
-            // If the detail pane was visible, then use the nestedNavController navigate call
-            // directly
-            nestedNavController.navigateToTopic(topicId) {
-                popUpTo<DetailPaneNavHostRoute>()
+        nestedNavHostStartRoute = TopicRoute(id = topicId)
```

Comment:
Since we no longer have a nested NavHost, this could be renamed to `startRoute` or maybe just `route`.

Reply:
Ah definitely, renamed to `topicRoute` in the latest commit